transhumandesign/kag-base
# Reencode musics at a lower bitrate

## Status

**READY**: this PR is (to the best of your knowledge) ready to be incorporated into the game.

## Description

Music in KAG was encoded at a needlessly high bitrate. We can save ~15MiB by reencoding it to a lower bitrate, at no noticeable quality cost.

Before:
```
./ambient_cavern_creatures1.ogg: Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps, created by: Xiph.Org libVorbis I (1.2.0)
./ambient_cavern_creatures2.ogg: Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps, created by: Xiph.Org libVorbis I (1.2.0)
./ambient_cavern_creatures3.ogg: Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps, created by: Xiph.Org libVorbis I (1.2.0)
./ambient_cavern_creatures4.ogg: Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps, created by: Xiph.Org libVorbis I (1.2.0)
./ambient_cavern.ogg:            Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps, created by: Xiph.Org libVorbis I (1.2.0)
./ambient_forrest.ogg:           Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps, created by: Xiph.Org libVorbis I (1.2.0)
./ambient_mountain.ogg:          Ogg data, Vorbis audio, mono, 48000 Hz, ~96000 bps, created by: Xiph.Org libVorbis I (1.2.0)
./KAGWorld1-10a.ogg:             Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-11a.ogg:             Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-12a.ogg:             Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-13intro.ogg:         Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-13+Intro.ogg:        Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-13.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-14.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-14outro.ogg:         Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-14+Outro.ogg:        Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-1a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-2a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-3a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-4a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-5a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-6a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-7a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-8a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorld1-9a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorldIntroA.ogg:            Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorldIntroShortA.ogg:       Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./KAGWorldQuickOut.ogg:          Ogg data, Vorbis audio, stereo, 44100 Hz, ~499821 bps, created by: Xiph.Org libVorbis I (1.0)
./world_intro.ogg:               Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps, created by: Xiph.Org libVorbis I (1.2.0)
```

After:
```
$ file ./Base/Sounds/Music/*.ogg
./ambient_cavern_creatures1.ogg: Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps
./ambient_cavern_creatures2.ogg: Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps
./ambient_cavern_creatures3.ogg: Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps
./ambient_cavern_creatures4.ogg: Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps
./ambient_cavern.ogg:            Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps
./ambient_forrest.ogg:           Ogg data, Vorbis audio, mono, 44100 Hz, ~96000 bps
./ambient_mountain.ogg:          Ogg data, Vorbis audio, mono, 48000 Hz, ~96000 bps
./KAGWorld1-10a.ogg:             Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-11a.ogg:             Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-12a.ogg:             Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-13intro.ogg:         Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-13+Intro.ogg:        Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-13.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-14.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-14outro.ogg:         Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-14+Outro.ogg:        Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-1a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-2a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-3a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-4a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-5a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-6a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-7a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-8a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorld1-9a.ogg:              Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorldIntroA.ogg:            Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorldIntroShortA.ogg:       Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./KAGWorldQuickOut.ogg:          Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
./world_intro.ogg:               Ogg data, Vorbis audio, stereo, 44100 Hz, ~160000 bps
```

## Steps to Test or Reproduce

N/A, just listen at the music.

If you want to compare audio you can do so by blind testing the difference. I noticed none even with a high volume and good quality headphones.

You can check the waveform difference using Audacity:

1. File -> Import -> Audio the old file
2. File -> Import -> Audio with the new file
3. Select the waveform of the new track
4. Effects -> Invert
5. Tracks -> Mix -> Mix and render
6. Select the waveform of the mixed track
7. Analyze -> Plot spectrum

The spectrum plot should show that the difference cannot be heard.
